### ISSUE TITLE
False-positive "Nothing to Document" error

### ISSUE THREAD

**Issue by happicamper** [UWHealth/markdown-documentation-generator#3]
Hello, I'm having a problem installing markdown-documentation-generator. 
Whenever I successfully initialized `md_documentation init` command and when I issue the command `md_documentation init` itself I'm producing this error:
```
[Style Guide] Reading scss, sass, less, md files...

[Style Guide] Could not find anything to document.
 Please check the following:
  * You've used /*SG*/ style comments.
  * Your "rootFolder" setting is pointing to the root of your style guide files.
  * If you're using the default settings, try using the "init" argument.
 If you're still receiving this error, please check the documentation or file an issue at github.com/UWHealth/markdown-documentation-generator/
```
And please tell me what should I do next to publish my documents. Thank you in advance!

**Comment by Cleecanth** [comment]
It sounds like you've successfully created your `.styleguide` file (which should now live in the root of your project), so you should be able to just run `md_documentation`.

The error you're receiving means no properly formatted comments are found in the scss, sass, less, or md files scanned. Are these the file types you want scanned? If not, change the settings under "fileExtensions". 

Without more detail, it's hard to say what's going on.

If you could post your project structure and settings in your .styleguide file, I should be able to help you further.

**Comment by happicamper** [comment]
Thank you for your response @Cleecanth! 
Can you give me a properly formatted working markdown file? I only configured the `.styleguide` file under `fileExtensions` to : 
```
"fileExtensions": {
		"scss": false,
		"sass": false,
		"less": false,
		"md": true,
		"css": false
	},
```
Thank you for your help, really appreciate it.

**Comment by Cleecanth** [comment]
It looks like you're only using markdown to document, which is fine but instead of CSS comments, you'll need to use specialty tags ([documentation here](https://github.com/UWHealth/markdown-documentation-generator#markdown-files)).

It'd look something like this:

```md
<sg>
# Category/Article

Some comments about this thing.
</sg>

<sg>
# Category/Another Article

Some more comments about this other thing.
</sg>
```

It's worth noting that since you're using markdown exclusively, it might make more sense to use a straight markdown formatter rather than this particular tool.

**Comment by happicamper** [comment]
Hello again, I've used the special tag `<sg></sg>` also tried your exact example above but it seems I'm experiencing the same error: 
```
[Style Guide] Reading md files...

[Style Guide] Could not find anything to document.
 Please check the following:
  * You've used /*SG*/ style comments.
  * Your "rootFolder" setting is pointing to the root of your style guide files.
  * If you're using the default settings, try using the "init" argument.
 If you're still receiving this error, please check the documentation or file an issue at github.com/UWHealth/markdown-documentation-generator/
```
Thank you so much for trying to help.

**Comment by lesleyblankendal** [comment]
@happicamper did you get it working? I am having the same issue even with the example comment posted by cleecanth. Ran init and have a folder with only .styleguide, test.md, test.scss to check if it was a problem with my md file.

Must be something simple I am overlooking...

**Comment by Cleecanth** [comment]
@lesleyblankendal — would you mind posting the contents of your .styleguide, test.md, and test.scss files? Should be able to help you from there.

It's also worth running `md_documentation lf` just to see what files are being read.

**Comment by happicamper** [comment]
@lesleyblankendal Unfortunately I didn't manage it to work. I've moved to other generators, I make this one pending issue so that maybe in the future it might be answered.

**Comment by lesleyblankendal** [comment]
with lf command
[Style Guide] Reading test.scss
[Style Guide] Reading test.md

content from style guide 
```
{
	"sgComment": "SG",
	"exampleIdentifier": "html_example",
	"sortCategories": true,
	"sections": {
		"styles": "",
		"development": "Dev:"
	},
	"rootFolder": "./",
	"excludeDirs": [
		"target",
		"node_modules",
		".git"
	],
	"fileExtensions": {
		"scss": true,
		"sass": true,
		"less": true,
		"md": true,
		"css": false
	},
	"templateFile": "../../../../../../usr/local/lib/node_modules/markdown-documentation-generator/template/template.hbs",
	"themeFile": "../../../../../../usr/local/lib/node_modules/markdown-documentation-generator/template/theme.css",
	"htmlOutput": "./styleguide/styleguide.html",
	"jsonOutput": "./styleguide/styleguide.json",
	"handlebarsPartials": {
		"jquery": "../../../../../../usr/local/lib/node_modules/markdown-documentation-generator/template/jquery.js",
		"sticky": "../../../../../../usr/local/lib/node_modules/markdown-documentation-generator/template/sticky.js"
	},
	"highlightStyle": "arduino-light",
	"highlightFolder": "../../../../../../usr/local/lib/node_modules/markdown-documentation-generator/node_modules/highlight.js/styles",
	"customVariables": {
		"pageTitle": "Style Guide"
	},
	"markedOptions": {
		"gfm": true,
		"breaks": true,
		"smartypants": true
	}
}
```

content from test.md
```
<sg>
# Category/Article

Some comments about this thing.
</sg>

<sg>
# Category/Another Article

Some more comments about this other thing.
</sg>
```

content from test.scss
syntax gets a bit tripped up by the backticks in the file but I copy pasted it from the readme https://github.com/UWHealth/markdown-documentation-generator#comment-style

**Comment by Cleecanth** [comment]
I was finally able to repeat this issue and I've got a fix in the works. 

There is essentially a race condition that occurs when documentation is occurring on extra small project folders. Ultimately, the file contents aren't being read before the file walker has finished processing the file tree. 

Should have a fix out in the next day.

**Comment by Cleecanth** [comment]
Update your version to 2.4.1. Problem should be resolved.

Fixed via #4
